katsdpcontroller/product_config.py

```diff
@@ -59,6 +59,18 @@ def _url_n_endpoints(url: Union[str, yarl.URL]) -> int:
     return len(endpoint_list_parser(None)(url.host))
 
 
+def _make_antenna(description: str) -> katpoint.Antenna:
+    """Create an antenna object from a description.
+
+    The :class:`katpoint.Antenna` constructor can throw assorted errors, so this
+    helper method always raises :exc:`ValueError` if the description is invalid.
+    """
+    try:
+        return katpoint.Antenna(description)
+    except Exception as exc:
```

Comment:
Just catching an `Exception` seems risky to me. Doesn't katpoint already raise a `ValueError` if the `description` string is bad?

Reply:
That's the problem: there are assorted errors that can occur (I don't recall exactly what, since this is code I wrote long ago and refactored here).

Reply:
I suppose catching and re-raising it again like this works then.
But you lose the original type, you raise it as a `ValueError`. You do preserve the message though I guess.

Reply:
Actually cancel that. I went to play around with what this does if you use `raise .. from .. ` and the behaviour is sane. You see both exceptions.

Reply:
Yes, losing the original type is the point. The katpoint interface is nasty because you get an exception of some arbitrary type depending on what it was trying to do (maybe you get an IndexError if you didn't pass enough fields or something), but I want the product_config constructors to predictably raise ValueError if they're given bad input.